shared/naturalcrit/splitPane/splitPane.jsx

```diff
@@ -1,200 +1,121 @@
+		return ()=>window.removeEventListener('resize', handleResize);
+	}, [storageKey]);
+
+	const limitPosition = (x, min = 1, max = window.innerWidth - 13)=>{
+		return Math.round(Math.min(max, Math.max(min, x)));
+	};
+
+	const handleUp =(e)=>{
 		e.preventDefault();
-		if(this.state.isDragging){
-			this.props.onDragFinish(this.state.currentDividerPos);
-			window.localStorage.setItem(this.props.storageKey, this.state.currentDividerPos);
+		if(isDragging) {
+			onDragFinish(dividerPos);
+			window.localStorage.setItem(storageKey, dividerPos);
 		}
-		this.setState({ isDragging: false });
-	},
+		setIsDragging(false);
+	};
 
-	handleDown : function(e){
+	const handleDown = (e)=>{
 		e.preventDefault();
-		this.setState({ isDragging: true });
-		//this.unFocus()
-	},
-
-	handleMove : function(e){
-		if(!this.state.isDragging) return;
+		setIsDragging(true);
+	};
 
+	const handleMove = (e)=>{
+		if(!isDragging) return;
 		e.preventDefault();
-		const newSize = this.limitPosition(e.pageX);
-		this.setState({
-			currentDividerPos : newSize,
-			userSetDividerPos : newSize
-		});
-	},
-
-	liveScrollToggle : function() {
-		window.localStorage.setItem('liveScroll', String(!this.state.liveScroll));
-		this.setState({ liveScroll: !this.state.liveScroll });
-	},
-	/*
-	unFocus : function() {
-		if(document.selection){
-				document.selection.empty();
-		}else{
-			window.getSelection().removeAllRanges();
-		}
-	},
-	*/
-
-	setMoveArrows : function(newState) {
-		if(this.state.showMoveArrows != newState){
-			this.setState({
-				showMoveArrows : newState
-			});
-		}
-	},
-
-	renderMoveArrows : function(){
-		if(this.state.showMoveArrows) {
-			return <>
-				<div className='arrow left'
-					style={{ left: this.state.currentDividerPos-4 }}
-					onClick={()=>this.setState({ moveSource: !this.state.moveSource })} >
-					<i className='fas fa-arrow-left' />
-				</div>
-				<div className='arrow right'
-					style={{ left: this.state.currentDividerPos-4 }}
-					onClick={()=>this.setState({ moveBrew: !this.state.moveBrew })} >
-					<i className='fas fa-arrow-right' />
-				</div>
-				<div id='scrollToggleDiv' className={this.state.liveScroll ? 'arrow lock' : 'arrow unlock'}
-					style={{ left: this.state.currentDividerPos-4 }}
-					onClick={this.liveScrollToggle} >
-					<i id='scrollToggle' className={this.state.liveScroll ? 'fas fa-lock' : 'fas fa-unlock'} />
-				</div>
-			</>;
-		}
-	},
-
-	renderDivider : function(){
-		return <>
-			{this.props.showDividerButtons && this.renderMoveArrows()}
-			<div className='divider' onPointerDown={this.handleDown} >
-				<div className='dots'>
-					<i className='fas fa-circle' />
-					<i className='fas fa-circle' />
-					<i className='fas fa-circle' />
+		const newSize = limitPosition(e.pageX);
+		setDividerPos(newSize);
+	};
+
+	const liveScrollToggle = ()=>{
+		window.localStorage.setItem('liveScroll', String(!liveScroll));
+		setLiveScroll(!liveScroll);
+	};
+
+	const moveArrows = showMoveArrows && (
```

Comment:
Let's continue to call this `renderMoveArrows` for consistency with all the other functions that output JSX.